packages/agents-hosting/src/auth/msalTokenProvider.ts

```diff
@@ -51,6 +51,22 @@ export class MsalTokenProvider implements AuthProvider {
     return token
   }
 
+  public async acquireTokenOnBehalfOf (authConfig: AuthConfiguration, scopes: string[], oboAssertion: string): Promise<string> {
+    const cca = new ConfidentialClientApplication({
+      auth: {
+        clientId: authConfig.clientId as string,
+        authority: `https://login.microsoftonline.com/${authConfig.tenantId || 'botframework.com'}`,
```

Comment:
We need to make this configurable to support the sovereign Entra Deployments. 
in .net MSAL confidentialClientApplication can support this,  
in .net AgentsSDK we expose this as a setting to allow a user to override it. 
Unfortunately there does not seem to be single table for sovereign / national clouds, here is the starting point: https://learn.microsoft.com/en-us/entra/identity-platform/authentication-national-cloud#azure-ad-authentication-endpoints

I would recommend that we offer this URL as a setting to let the user provide it.